### ISSUE TITLE
Regression on master branch: false-positive error for enum mixin data types

### ISSUE THREAD

**Issue by AlexWaygood** [python/mypy#12258]
Using the current master branch, mypy reports a false positive error for the following snippet of code:

```python
from enum import Enum

class MyEnum(Enum): ...

class BytesEnum(bytes, MyEnum): ...  # error: Only a single data type mixin is allowed for Enum subtypes, found extra "enum_test.MyEnum"
```

The error is not reported using mypy 0.931.

## Expected behaviour

There should be no error reported; this works fine at runtime (and this inheritance structure is in fact used within `enum.py` in the CPython stdlib).

Related:

- #12220
- https://github.com/python/typeshed/pull/7389

cc. @sobolevn

**Comment by hauntsaninja** [comment]
Maybe we revert #12026 ?
